indexer/src/types/transaction.ts

```diff
@@ -67,96 +71,181 @@ export function toEthTx({
 }
 
 /**
- * @param typeTransaction - Typed transaction to be converted.
- * @param receipt The transaction receipt of the transaction.
+ * Calculates the chain ID from the transaction.
+ *
+ * @param typedTransaction - The typed transaction object.
+ * @param jsonTx - The JSON representation of the transaction.
+ * @returns - The chain ID as a string, or undefined if the chain ID cannot be determined.
+ *
+ * If the transaction is a legacy transaction that supports EIP-155 replay protection,
+ * the chain ID is calculated from the `v` value using the formula:
+ * v = 35 + 2 * chainId + yParity -> chainId = (v - 35) / 2
+ */
+function chainId(
+  typedTransaction: TypedTransaction,
+  jsonTx: JsonTx,
+): string | undefined {
+  const { v, chainId } = jsonTx;
+
+  if (
+    isLegacyTx(typedTransaction) &&
+    typedTransaction.supports(Capability.EIP155ReplayProtection)
+  ) {
+    return v
+      ? bigIntToHex((BigInt(v) - 35n) / 2n)
+      : (console.error("jsonTx.v is undefined"), undefined);
+  }
+  return chainId;
+}
+
+/**
+ * Adds the yParity field to the transaction result if it is an EIP-1559 or EIP-2930 transaction.
+ *
+ * @param typedTransaction - The typed transaction object.
+ * @param jsonTx - The JSON representation of the transaction.
+ * @param result - The transaction result object in Ethereum format.
+ */
+function setYParityFlag(
+  typedTransaction: TypedTransaction,
+  jsonTx: JsonTx,
+  result: ExtendedJsonRpcTx,
+): void {
+  // Check if the transaction is an EIP-1559 or EIP-2930 transaction
+  if (
+    isFeeMarketEIP1559TxData(typedTransaction) ||
+    isAccessListEIP2930Tx(typedTransaction)
+  ) {
+    // Add the yParity field to the result
+    result.yParity = jsonTx.v;
+  }
+}
+
+/**
+ * Adds the isRunOutOfResources flag to the transaction result if the transaction
+ * was reverted due to running out of resources.
+ *
+ * @param receipt - The transaction receipt object.
+ * @param result - The transaction result object in Ethereum format.
+ */
+function flagRunOutOfResources(
```

Comment:
same here, I think the `set` is more explicit @tcoratger what do you think?

Reply:
yes let us put the `set` prefix in both cases